presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -827,13 +830,18 @@ public void pushdownFilterAndProjection(PageSourceOptions options, int[] channel
             nonConstantFilterFunctions = this.filterFunctions.toArray(new AbstractFilterFunction[0]);
         }
 
+        int[] channels = new int[outputColumns.size()];
+        for (int i = 0; i < outputColumns.size(); i++) {
+            channels[i] = outputColumns.get(i) == -1 ? -1 : i;
+        }
+
         reader = new ColumnGroupReader(
                 streamReaders,
                 presentColumns,
                 channelColumns,
                 types,
-                options.getInternalChannels(),
-                options.getOutputChannels(),
+                channels,
```

Comment:
I saw the only two callers of ColumnGroupReader constructor are passing in the same set of chanels as internalChannels and outputChannels. If there're no case these can be different, can we simplify the interface of ColumnGroupReader() to take one instead of two input parameters?

Reply:
Yes. This can be done. I'd like to defer this to a future PRs, because, all of the connector implementation needs to be revisited.